portal-impl/src/com/liferay/portal/util/PortalImpl.java

```diff
@@ -4424,16 +4424,29 @@ else if (doAsGroup != null) {
 				if (liveGroup.isStaged() &&
 					!liveGroup.isStagedPortlet(portletId)) {
 
-					Layout liveGroupLayout =
-						LayoutLocalServiceUtil.fetchLayoutByUuidAndGroupId(
-							layout.getUuid(), liveGroup.getGroupId(),
-							layout.isPrivateLayout());
+					long portletScopeGroupId = _getScopeGroupId(
+						themeDisplay, layout, portletId);
 
-					if ((liveGroupLayout != null) &&
-						liveGroupLayout.hasScopeGroup()) {
+					if (portletScopeGroupId != layout.getGroupId()) {
+						Group portletScopeGroup =
+							GroupLocalServiceUtil.fetchGroup(
+								portletScopeGroupId);
 
-						scopeGroupId = _getScopeGroupId(
-							themeDisplay, liveGroupLayout, portletId);
+						if (portletScopeGroup.isStagingGroup()) {
+							if (!checkStagingGroup) {
+								Group portletScopeLiveGroup =
+									portletScopeGroup.getLiveGroup();
+
+								scopeGroupId =
+									portletScopeLiveGroup.getGroupId();
+							}
+							else {
+								scopeGroupId = portletScopeGroupId;
+							}
+						}
+						else {
+							scopeGroupId = portletScopeGroup.getGroupId();
+						}
 					}
```

Comment:
Hi @DaniSzimko，for the fix, it seems a little complexed. We can simplify the logic. After `long portletScopeGroupId = _getScopeGroupId()`,  two else blocks actually return the same `portletScopeGroupId`. We may simplify it as the foolowing:
```
scopeGroupId = _getScopeGroupId(themeDisplay, layout, portletId);

					if (!checkStagingGroup && scopeGroupId != layout.getGroupId()) {
						Group portletScopeGroup =
								GroupLocalServiceUtil.fetchGroup(scopeGroupId);

						if (portletScopeGroup.isStagingGroup()) {
							Group portletScopeLiveGroup =
								portletScopeGroup.getLiveGroup();

							scopeGroupId = portletScopeLiveGroup.getGroupId();
						}
					}
					else if (checkStagingGroup &&
							 !liveGroup.isStagedRemotely()) {

						Group stagingGroup = liveGroup.getStagingGroup();

						scopeGroupId = stagingGroup.getGroupId();
					}
					else {
						scopeGroupId = liveGroup.getGroupId();
					}
```

But it seems also be not good.